diff --git a/src/Publisher/FilesystemPublisher.php b/src/Publisher/FilesystemPublisher.php
index ac68234..aed234c 100644
--- a/src/Publisher/FilesystemPublisher.php
+++ b/src/Publisher/FilesystemPublisher.php
@@ -213,7 +213,7 @@ class FilesystemPublisher extends Publisher
             }
             $fullPath = $dir . DIRECTORY_SEPARATOR . $fileOrDir;
             // we know html will always be generated, this prevents double ups
-            if (is_file($fullPath) && pathinfo($fullPath)['extension'] == 'html') {
+            if (is_file($fullPath) && pathinfo($fullPath, PATHINFO_EXTENSION) == 'html') {
                 $result[] = $this->pathToURL($fullPath);
                 continue;
             }
